### DISCUSSION TITLE
Query collections, filtered by metafield value

### DISCUSSION THREAD

**Discussion by patjakubik** [Shopify/storefront-api-feedback#113] [Ideas & Requests]
We would love to have ability to query collections, filtered by metafield value.

Use case:

We have different types of collections, e.g., "brand" or "designer". We would like to have pages "All brands", "All designers" that lists all collections of particular type. 

Before headless, we used collection templates, but this is obviously not available in Storefront API. Having "collection type" metafield seem to be the only alternative here.

**Comment by vixdug** [comment]
You are able to query collections and filter based on product metafields: https://shopify.dev/api/examples/filter-products#query-products-by-metafield-value. Alternatively, if you are looking to filter multiple collections based on metafield you may want to do something like: 

```
{
  collectionByHandle(handle: "all") {
    id
    metafield(namespace: "global", key: "designer") {
      namespace
      key
      value
    }
  }
}
```

**Reply by patjakubik** [reply]
Hey @vixdug thanks for response. I meant being able to query collections with metafields, not the querying the products with metafields. See example query below.

```
query {
	collections(
		metafield: { namespace: "global", key: "type", value: "designer" }
	) {
		nodes {
			id
			title
		}
	}
}
```
